### DISCUSSION TITLE
Benefit to using Polly with AWS SQS?

### DISCUSSION THREAD

**Discussion by lonix1** [App-vNext/Polly#2370] [Q&A] [answered]
Hi @martincostello, looking for advice if you have a moment. I discovered you're the main guy behind https://github.com/justeattakeaway/JustSaying too, so you're the expert in this matter.

I use Polly for rate limiting AWS/Azure services which have quotas (e.g. SES) or for retries and timeouts.

I recently started using SQS and was wondering whether there's any benefit to using Polly somehow?

The SDK's SQS client already has built-in timeouts and retries, however Polly may have strategies which are useful to use with SQS. Since I'm new to it, I don't have the experience to know whether that's the case.

Basically, do **you** use Polly with SQS? Any thoughts appreciated. :pray:

**Comment by martincostello** [comment, marked as answer]
Hey there - we don't typically use Polly with the AWSSDK _directly_, but we often do in the business logic that consumes it (for example to retry a fetch-compute-publish block of code).

We wouldn't typically guard AWS calls with rate-limiting strategies - the most common catch-all scenario we use is for circuit breaking. That way, when an AWS outage occurs, we fail fast rather than keep hammering AWS with retries.

**Reply by lonix1** [reply]
You've experienced AWS outages? Wow, a black swan event indeed! :smile:
